src/java.base/share/classes/java/util/DualPivotQuicksort.java

```diff
@@ -58,163 +58,150 @@ final class DualPivotQuicksort {
      */
-    private static final int MAX_INSERTION_SORT_SIZE = 44;
+    private static final int MAX_INSERTION_SORT_SIZE = 51;
 
-    /**
-     * Min array size to perform sorting in parallel.
-     */
-    private static final int MIN_PARALLEL_SORT_SIZE = 4 << 10;
+    /* ---------------------- Merging sort ---------------------- */
 
     /**
-     * Min array size to try merging of runs.
+     * Min size of array to use merging sort.
      */
-    private static final int MIN_TRY_MERGE_SIZE = 4 << 10;
+    private static final int MIN_MERGING_SORT_SIZE = 512;
 
     /**
-     * Min size of the first run to continue with scanning.
+     * Min size of run to continue scanning.
      */
-    private static final int MIN_FIRST_RUN_SIZE = 16;
+    private static final int MIN_RUN_SIZE = 64;
 
     /**
-     * Min factor for the first runs to continue scanning.
+     * Max capacity of the index array to track the runs.
      */
-    private static final int MIN_FIRST_RUNS_FACTOR = 7;
+    private static final int MAX_RUN_CAPACITY = 10 << 10;
 
-    /**
-     * Max capacity of the index array for tracking runs.
-     */
-    private static final int MAX_RUN_CAPACITY = 5 << 10;
+    /* --------------------- Numerical sort --------------------- */
 
     /**
-     * Min number of runs, required by parallel merging.
+     * Min size of array to use Radix sort.
      */
-    private static final int MIN_RUN_COUNT = 4;
+    private static final int MIN_RADIX_SORT_SIZE = 640;
 
     /**
-     * Min array size to use parallel merging of parts.
+     * Min size of array to use counting sort.
      */
-    private static final int MIN_PARALLEL_MERGE_PARTS_SIZE = 4 << 10;
+    private static final int MIN_COUNTING_SORT_SIZE = 9 << 10;
 
-    /**
-     * Min size of a byte array to use counting sort.
-     */
-    private static final int MIN_BYTE_COUNTING_SORT_SIZE = 64;
+    /* --------------------- Parallel sort ---------------------- */
 
     /**
-     * Min size of a short or char array to use counting sort.
+     * Min size of array to perform sorting in parallel.
      */
-    private static final int MIN_SHORT_OR_CHAR_COUNTING_SORT_SIZE = 1750;
+    private static final int MIN_PARALLEL_SORT_SIZE = 3 << 10;
+
+    /* --------------------- Infrastructure --------------------- */
 
     /**
-     * Threshold of mixed insertion sort is incremented by this value.
+     * Max recursive depth before switching to heap sort.
      */
-    private static final int DELTA = 3 << 1;
+    private static final int MAX_RECURSION_DEPTH = 64 << 1;
 
     /**
-     * Max recursive partitioning depth before using heap sort.
+     * Max size of additional buffer in bytes,
+     *      limited by max_heap / 16 or 2 GB max.
      */
-    private static final int MAX_RECURSION_DEPTH = 64 * DELTA;
+    private static final int MAX_BUFFER_SIZE =
+        (int) Math.min(Runtime.getRuntime().maxMemory() >>> 4, Integer.MAX_VALUE);
```

Comment:
A small suggestion: since Java 25, you may utilize `Math.clamp` to avoid explicit `(int)` cast:

```java
private static final int MAX_BUFFER_SIZE =
        Math.clamp(Runtime.getRuntime().maxMemory() >>> 4, 0, Integer.MAX_VALUE);
```